File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {

        if(head == null || head.next == null){
            return head;
        }
        ListNode prev = null;
        ListNode f = head;
        ListNode s = f.next;

        while(f != null){
            f.next = prev;
            prev = f;
            f = s;
            if(s != null){
                s = s.next;
            }
        }
        head = prev;
        return head;
    }    
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(head == null || head.next == null){
        return head;
    }

    let prev = null;
    let f = head;
    let s = f.next;


    while(f!= null){
        f.next = prev;
        prev = f;
        f = s;
        if(s!== null){
            s = s.next;
        }
    }
    head = prev;
    return head;
};